txtorcon/circuit.py

```diff
@@ -275,3 +275,25 @@ def __str__(self):
         path = ' '.join([x.ip for x in self.path])
         return "<Circuit %d %s [%s] for %s>" % (self.id, self.state, path,
                                                 self.purpose)
+
+class CircuitBuildTimedOutError(Exception):
+    """
+    This exception is thrown when using `timed_circuit_build`
+    and the circuit build times-out.
+    """
+
+def build_timeout_circuit(tor_state, clock, path, timeout):
+    """
```

Comment:
yeah clock could be called reactor. yeah could make this an optional timeout for build_circuit... but does it already have a reactor reference?